src/mgmt/srv.c

```diff
@@ -310,17 +292,8 @@ ds_mgmt_mark_hdlr(crt_rpc_t *rpc)
 
 	tc_in->m_mark = in->m_mark;
 	rc = dss_rpc_send(tc_req);
-	if (rc != 0) {
-		crt_req_decref(tc_req);
-		D_GOTO(out, rc);
-	}
 
-	out = crt_reply_get(tc_req);
-	rc = out->m_rc;
-	if (rc != 0) {
-		crt_req_decref(tc_req);
-		D_GOTO(out, rc);
-	}
+	crt_req_decref(tc_req);
```

Comment:
the change here looks good to me.
But just checked that inside dss_rpc_send() it take one extra ref ("L40 crt_req_addref(rpc);), and no place to dec that ref (seems should decref in its complete cb - rpc_cb(), but it did not do that)?

Reply:
OK, i see, "dss_rpc_send() it take one extra ref " that one is to ensure the rpc is valid after completion.